src/loaders/compute_tools/tool_common.py

```diff
@@ -617,6 +610,13 @@ def create_fatal_tuple(
     return fatal_tuple
 
 
+def get_threads(system_utilization: float, threads: int = None) -> int:
```

Comment:
The tool_common file should only contain helper methods that are exclusive to tools. If a method is used by other parts of the pipeline it should go in loader helpers

Reply:
then I assume threading stuff should be put into loader_helpers instead of ncbi helper?

Reply:
Hmm... if you can make it generic enough for reuse in other places sure. That being said, AFAIK there's nothing else making use of it right now (or am I wrong about that?) and so it could live in the ncbi code for now and we could do the refactoring when we actually need it and have a real use case for figuring out which parts are sharable and which aren't

Reply:
tool_common.py is also using it for sure.

Reply:
In the same or very close way? If so then yeah, it'd make sense to move it into the common loader code or its own module

Reply:
nah, I feel they are kinda diff in terms of tool_common.py has more data processing in between get_ threads() and multiprocessing.Pool call. Also multiprocessing.Pool call is nested inside _execute() func as well. Not quite sure how can I make it generic enough that can reuse in both places other than get_threads().

Reply:
Ok, I'd leave it in the ncbi code for now then

Reply:
Just to make sure we are on the same page, we are not changing multithreading stuff in gtdb.py file for now.

Reply:
All the multithreading code should move from gtdb.py to the ncbi helper so it can be reused when we write another script that downloads stuff from ncbi

Reply:
Updated!

Reply:
:+1: